model/src/main/kotlin/utils/DatabaseUtils.kt

```diff
@@ -36,6 +38,11 @@ import org.ossreviewtoolkit.utils.ort.ORT_FULL_NAME
 import org.ossreviewtoolkit.utils.ort.log
 
 object DatabaseUtils {
+    /**
+     * This map holds the [HikariDataSource] based on the [PostgresConfiguration].
+     */
+    private val dataSources = ConcurrentHashMap<PostgresConfiguration, Lazy<DataSource>>()
```

Comment:
FYI, the original reason why the `DataSource` was wrapped in `Lazy` in 32a1f0724ca3e0aefb31654f3908152ecc36906c was that Hikari establishes the connection very early, originally upon reading the config, even if no database connection was required in the executes code path. That's inconvenient for development if you don't have the data source configured in `ort.conf` (because you shouldn't need to). I'd be glad if this PR doesn't change this.

Reply:
I agree here with Sebastian and also due to the reasoning  [here](https://github.com/oss-review-toolkit/ort/pull/5496#discussion_r908385141), I will keep the `DataSource` wrapped in a Lazy object.